File: Models/PollingChartRenderFrame.vb
Namespace Models
    Public Class PollingChartRenderFrame
        Private ReadOnly _rawCurrentRate As Double
        Private ReadOnly _historyPoints As IReadOnlyList(Of PollingHistoryPoint)

        Public Sub New(rawCurrentRate As Double, historyPoints As IReadOnlyList(Of PollingHistoryPoint))
            _rawCurrentRate = rawCurrentRate
            _historyPoints = If(historyPoints, Array.Empty(Of PollingHistoryPoint)())
        End Sub

        Public ReadOnly Property RawCurrentRate As Double
            Get
                Return _rawCurrentRate
            End Get
        End Property

        Public ReadOnly Property HistoryPoints As IReadOnlyList(Of PollingHistoryPoint)
            Get
                Return _historyPoints
            End Get
        End Property
    End Class
End Namespace

File: src/ClickSyncMouseTester/Models/PollingChartRenderFrame.cs
using System;
using System.Collections.Generic;

namespace ClickSyncMouseTester.Models;

public class PollingChartRenderFrame
{
    private readonly double _rawCurrentRate;

    private readonly IReadOnlyList<PollingHistoryPoint> _historyPoints;

    public double RawCurrentRate => _rawCurrentRate;

    public IReadOnlyList<PollingHistoryPoint> HistoryPoints => _historyPoints;

    public PollingChartRenderFrame(double rawCurrentRate, IReadOnlyList<PollingHistoryPoint> historyPoints)
    {
        _rawCurrentRate = rawCurrentRate;
        _historyPoints = historyPoints ?? Array.Empty<PollingHistoryPoint>();
    }
}





